core/src/main/scala/org/apache/spark/status/api/v1/AllStagesResource.scala

```diff
@@ -237,7 +237,8 @@ private[v1] object AllStagesResource {
   }
 
   def convertAccumulableInfo(acc: InternalAccumulableInfo): AccumulableInfo = {
-    new AccumulableInfo(acc.id, acc.name, acc.update, acc.value)
+    new AccumulableInfo(
+      acc.id, acc.name, acc.update.map(_.toString), acc.value.map(_.toString).orNull)
```

Comment:
yeah, I didn't want to introduce v2 just for this change. If you prefer we can do that separately